modules/mono/editor/GodotTools/GodotTools/Build/BuildProblemsView.cs

```diff
@@ -0,0 +1,683 @@
+
+                if (csvColumns.Length != 7)
+                {
+                    GD.PushError($"Expected 7 columns, got {csvColumns.Length}");
+                    continue;
+                }
+
+                var diagnostic = new BuildDiagnostic
+                {
+                    Type = csvColumns[0] switch
+                    {
+                        "warning" => BuildDiagnostic.DiagnosticType.Warning,
+                        "error" or _ => BuildDiagnostic.DiagnosticType.Error,
+                    },
+                    File = csvColumns[1],
+                    Line = int.Parse(csvColumns[2]),
+                    Column = int.Parse(csvColumns[3]),
+                    Code = csvColumns[4],
+                    Message = csvColumns[5],
+                    ProjectFile = csvColumns[6],
+                };
+
+                yield return diagnostic;
+            }
+        }
+
+        public void SetDiagnosticsFromFile(string csvFile)
+        {
+            var diagnostics = ReadDiagnosticsFromFile(csvFile);
+            SetDiagnostics(diagnostics);
+        }
+
+        public void SetDiagnostics(IEnumerable<BuildDiagnostic> diagnostics)
+        {
+            _diagnostics.Clear();
+            _problemCountByType.Clear();
+
+            _diagnostics.AddRange(diagnostics);
+            UpdateProblemsView();
+        }
+
+        public void Clear()
+        {
+            _problemsTree.Clear();
+            _diagnostics.Clear();
+            _problemCountByType.Clear();
+
+            UpdateProblemsView();
+        }
+
+        private void CopySelectedProblems()
+        {
+            var selectedItem = _problemsTree.GetNextSelected(null);
+            if (selectedItem == null)
+                return;
+
+            var selectedIdxs = new List<int>();
+            while (selectedItem != null)
+            {
+                int selectedIdx = (int)selectedItem.GetMetadata(0);
+                selectedIdxs.Add(selectedIdx);
+
+                selectedItem = _problemsTree.GetNextSelected(selectedItem);
+            }
+
+            if (selectedIdxs.Count == 0)
+                return;
+
+            var selectedDiagnostics = selectedIdxs.Select(i => _diagnostics[i]);
+
+            var sb = new StringBuilder();
+
+            foreach (var diagnostic in selectedDiagnostics)
+            {
+                if (!string.IsNullOrEmpty(diagnostic.Code))
+                    sb.Append($"{diagnostic.Code}: ");
+
+                sb.AppendLine($"{diagnostic.Message} {diagnostic.File}({diagnostic.Line},{diagnostic.Column})");
+            }
+
+            string text = sb.ToString();
+
+            if (!string.IsNullOrEmpty(text))
+                DisplayServer.ClipboardSet(text);
+        }
+
+        private void ToggleLayout(bool pressed)
+        {
+            _layout = pressed ? ProblemsLayout.List : ProblemsLayout.Tree;
+
+            var editorSettings = GodotSharpEditor.Instance.GetEditorInterface().GetEditorSettings();
```

Comment:
Can be this, I think?

```suggestion
            var editorSettings = EditorInterface.Singleton.GetEditorSettings();
```